### ISSUE TITLE
Instructions on using rules_kotlin with bzlmod

### ISSUE THREAD

**Issue by kylepl** [bazelbuild/rules_kotlin#1060]
I see that bzlmod experimental support was just added in 1.9.0, very exciting!

When I try to start using Kotlin in my repo by just adding:
`bazel_dep(name = "rules_kotlin", version = "1.9.0")`

to my MODULE.bazel, and _nothing_ to my WORKSPACE file, I get:
```
error loading package 'whyle/experimental': at /home/user/.cache/bazel/_bazel_my_user/fd0c5957b74bb38c2ac114f24487fc53/external/io_bazel_rules_kotlin/kotlin/jvm.bzl:9:5: at /home/my_user/.cache/bazel/_bazel_my_user/fd0c5957b74bb38c2ac114f24487fc53/external/io_bazel_rules_kotlin/kotlin/internal/opts.bzl:16:5: Unable to find package for @io_bazel_rules_kotlin_configured//kotlin:opts.bzl: The repository '@io_bazel_rules_kotlin_configured' could not be resolved: Repository '@io_bazel_rules_kotlin_configured' is not defined.
```

I wasn't sure if I should add:
```
load("@io_bazel_rules_kotlin//kotlin:repositories.bzl", "kotlin_repositories")
kotlin_repositories()
```

to my WORKSPACE still. Tried it, and get [this same problem again](https://github.com/bazelbuild/rules_kotlin/issues/883).

So basically not sure if there is an issue (using Bazel 6.4.0) or if I'm holding it wrong.

**Comment by kylepl** [comment]
For reasons I don't understand, it seems like `bazel_dep(name = "rules_kotlin", version = "1.9.0")` is pulling down 1.7.0, despite the [BCR](https://github.com/bazelbuild/bazel-central-registry/blob/main/modules/rules_kotlin/1.9.0/source.json) reference clearly being 1.9.0.

This was deduced from manually inspecting `/home/my_user/.cache/bazel/_bazel_my_user/fd0c5957b74bb38c2ac114f24487fc53/external/io_bazel_rules_kotlin/src/main/starlark/core/repositories/initialize.bzl`, and seeing when it was from (though it required matching it against https://github.com/bazelbuild/rules_kotlin/blob/e5c457c508f21f43f1686ce9be928d2e2a4fd6f8/src/main/starlark/core/repositories/initialize.release.bzl).

Also just doing `ls /home/my_user/.cache/bazel/_bazel_my_user/fd0c5957b74bb38c2ac114f24487fc53/external/io_bazel_rules_kotlin`, does not even show a MODULE.bazel - which I do see if I download the BCR 1.9.0 .tar.gz directly.

I'm probably missing something.

**Comment by kylepl** [comment]
So, while I don't have a full resolution on the issue yet, my problem was that I was still referring to `io_bazel_rules_kotlin`, but by default bzlmod was using `rules_kotlin`. I believe it was defined in my case because of a proto import I was using, also defining `io_bazel_rules_kotlin` to an older version.

I'm closing this, and will make a new one after a bit more exploration. Leaving my investigation up in case it helps anyone else.

**Comment by kylepl** [comment]
And to get a basic version all-the-way working, I had to do:
`bazel_dep(name = "rules_kotlin", version = "1.9.0", repo_name="io_bazel_rules_kotlin")`
and _nothing_ in the WORKSPACE file. But the repo_name was required.
